libs/payments/cart/src/lib/checkout.service.ts

```diff
@@ -2,41 +2,121 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import { Injectable } from '@nestjs/common';
+import { EligibilityService } from '@fxa/payments/eligibility';
+import { PayPalManager, PaypalCustomerManager } from '@fxa/payments/paypal';
 import {
   StripeClient,
   StripeManager,
   StripeSubscription,
+  SubplatInterval,
+  TaxAddress,
 } from '@fxa/payments/stripe';
-import { PayPalManager, PaypalCustomerManager } from '@fxa/payments/paypal';
-import { Injectable } from '@nestjs/common';
-import { ResultCart } from './cart.types';
 import { CheckoutError, CheckoutPaymentError } from './checkout.error';
+import {
+  CartTotalMismatchError,
+  CartEligibilityMismatchError,
+  CartEmailNotFoundError,
+} from './cart.error';
+import { CartManager } from './cart.manager';
+import { ResultCart } from './cart.types';
+import { handleEligibilityStatusMap } from './cart.utils';
+import { ContentfulService } from '@fxa/shared/contentful';
+import { AccountManager } from '@fxa/shared/account/account';
+import { DEFAULT_LOCALE } from '@fxa/shared/l10n';
 
 @Injectable()
 export class CheckoutService {
   constructor(
     private stripeClient: StripeClient,
     private stripeManager: StripeManager,
     private paypalCustomerManager: PaypalCustomerManager,
-    private paypalManager: PayPalManager
+    private paypalManager: PayPalManager,
+    private cartManager: CartManager,
+    private eligibilityService: EligibilityService,
+    private contentfulService: ContentfulService,
+    private accountManager: AccountManager
   ) {}
 
   async prePaySteps(cart: ResultCart) {
-    // TODO:
-    // - If uid not present, create user stub account and update cart
-    // - Check if customer exists, create via stripe manager and update cart
-    // - Make sure shipping address of customer is updated with cart taxAddress
-    // - Check if customer is eligible for product in cart
-    // - Check if amount to be charged matches amount in cart
-    // - Cancel incomplete subscriptions to price user is subscribing to
-
-    const { stripeCustomerId } = cart;
-    if (!stripeCustomerId) {
-      // TODO: this is a placeholder and won't be necessary once the above todo items are done
-      throw new Error("Stripe customer doesn't exist");
+    let customer;
+    const taxAddress = cart.taxAddress as any as TaxAddress;
+
+    if (!cart.email) {
+      throw new CartEmailNotFoundError(cart.id);
+    }
+
+    // if uid not found, create account stub and update cart
+    if (!cart.uid) {
+      const uid = await this.accountManager.createAccountStub(
+        cart.email,
+        cart.version,
```

Comment:
`cart.version` and `verifierVersion` are not equivalent. 

In `fxa-auth-server` it appears to be a config value, but unfortunately I don't have much more info than that. Might be worth checking with FxA folks?

Reply:
@bbangert as you wrote the method for createAccountStub, how can I pass in `verifierVersion`?

Reply:
Hardcoded with 1, filed ticket FXA-9693, and added TODO comment as per Slack conversation.